charts/opentelemetry-collector/templates/daemonset.yaml

```diff
@@ -10,6 +10,18 @@ spec:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
       {{- include "opentelemetry-collector.component" . | nindent 6 }}
+  updateStrategy:
+    {{- if eq .Values.rollout.strategy "RollingUpdate" }}
+    rollingUpdate:
+      {{- if .Values.rollout.rollingUpdate.maxUnavailable }}
+      maxUnavailable: {{ .Values.rollout.rollingUpdate.maxUnavailable }}
+      {{- else if .Values.rollout.rollingUpdate.maxSurge }}
+      maxSurge: {{ .Values.rollout.rollingUpdate.maxSurge }}
+      {{- else }}
+      maxUnavailable: 25%
```

Comment:
Isn't the default value for maxUnavailable 1?

Reply:
https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/#performing-a-rolling-update

Indeed, the default applied by Kuberenetes when this isn't specified is `1`. I've opted here for a default in the chart of `25%` instead, as IMO that provides a better out-of-the-box behaviour for the Daemonset, which doesn't start hitting helm deployment timeouts as a cluster grows.

If the preference is to stick with the defaults mirroring the k8s default behaviour, and have users of the chart override that, I can update the changeset to reflect that.

Reply:
I don't agree that 25% is a safe default value for daemonset. In case of a bad upgrade, user would lose monitoring of 1/4 cluster instead of only one node.

Let's keep it to kubernetes default value which is 1 I believe.

Reply:
For the sake of keeping existing charts user deployments the same, I would vote for setting it to 1. @dmitryax what do you think?

Reply:
Updated to use `1` as the default

Reply:
The default is now dropped entirely from the chart with `.Values.rollout.rollingUpdate` being dumped out, and relies on kubernetes to apply the default